### PR TITLE
Postmortem improvements

### PR DESCRIPTION
**Pull request by mhightower83**
Improve divide by 0 reporting to point back at the divide function caller

Improve stack offset adjustments to be independent of `__wrap_system_restart_local()`'s  stack frame size.

### CODE REVIEW

File: cores/esp8266/core_esp8266_postmortem.cpp [hunk 1]
```diff
@@ -110,10 +110,27 @@ static void cut_here() {
     ets_putc('\n');
 }
 
-void __wrap_system_restart_local() {
-    register uint32_t sp asm("a1");
-    uint32_t sp_dump = sp;
-
+static void postmortem_report(uint32_t sp_dump) __attribute__((used));
```
Comment (mcspr) [inline cores/esp8266/core_esp8266_postmortem.cpp:113]: Is it dropped, somehow?
I don't see any errors on my side after removing this and adding `static` below

File: cores/esp8266/core_esp8266_postmortem.cpp [hunk 2]
```diff
@@ -174,17 +199,32 @@ void __wrap_system_restart_local() {
...
+        // 256 Exception vector reserves stack space
+        //     filled in by "C" wrapper handler
+        //  16 Handler level 1 - enable icache
+        //  64 Handler level 2 - exception report
+        offset = 64 + 16 + 256;
     }
     else if (rst_info.reason == REASON_WDT_RST) {
         offset = 0x10;
     }
+    else if (rst_info.reason == REASON_USER_SWEXCEPTION_RST) {
```
Comment (mcspr) [inline cores/esp8266/core_esp8266_postmortem.cpp:225]: Also update ^ `REASON_WDT_RST` to decimal?

**Review by mcspr — COMMENTED** [review]
lgtm based on repeating crashes from #8670

File: cores/esp8266/core_esp8266_postmortem.cpp (hunk 2 above)
Comment (mhightower83) [reply cores/esp8266/core_esp8266_postmortem.cpp:224]: done

File: cores/esp8266/core_esp8266_postmortem.cpp (hunk 1 above)
Comment (mhightower83) [reply cores/esp8266/core_esp8266_postmortem.cpp]: Updated. There is some scenario where I need this, but not here. :confused:

**Pull request merged**